### PR TITLE
Fixes #31095 - Remove config groups model

### PR DESCRIPTION
**Pull request by shiramax**
Added to the plugin in https://github.com/theforeman/foreman_puppet_enc/pull/12

### CODE REVIEW

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -731,7 +726,7 @@ def root_pass
     Setting[:root_pass]
   end
 
-  include_in_clone :config_groups, :host_config_groups, :host_classes, :host_parameters, :lookup_values
```
Comment (ezr-ondrej) [inline app/models/host/managed.rb]: Note to myself: should be tested.

File: app/models/hostgroup.rb [hunk 2]
```diff
@@ -65,16 +65,6 @@ class Hostgroup < ApplicationRecord
   scoped_search :on => :id, :complete_enabled => false, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
   # for legacy purposes, keep search on :label
   scoped_search :on => :title, :complete_value => true, :rename => :label
-  scoped_search :relation => :config_groups, :on => :name, :complete_value => true, :rename => :config_group, :only_explicit => true, :operators => ['= ', '~ '], :ext_method => :search_by_config_group
-
-  def self.search_by_config_group(key, operator, value)
```
Comment (ezr-ondrej) [inline app/models/hostgroup.rb:65]: Needs to get tested.

**Review by ezr-ondrej — APPROVED** [review]
I need to run the tests, but apart of it looks that matches what's been added.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
We are missing drop of methods in `HostCommon` introduced in: https://github.com/theforeman/foreman_puppet_enc/pull/12/files#diff-d282d0fe7461e12531f5252b3aec5d5ba1efdcedad1ddf9fb4a1d8ea36dfca40R15-R39

**Comment by shiramax** [comment]
@ezr-ondrej tests are failing due to this migration:
 https://github.com/ezr-ondrej/foreman/blob/develop/db/migrate/20140415032811_add_config_group_counter_defaults.rb 
Should I remove it from this PR?

**Comment by ezr-ondrej** [comment]
> @ezr-ondrej tests are failing due to this migration:
> https://github.com/ezr-ondrej/foreman/blob/develop/db/migrate/20140415032811_add_config_group_counter_defaults.rb
> Should I remove it from this PR?

Yeah, eighter drop the migration completely, or just remove the model manipulation :)
Doesn't matter too much as we will drop it in the migration drop PR :)

**Review by ezr-ondrej — APPROVED** [review]
Thanks @shiramax :+1:

**Pull request merged**